tensorflow/compiler/mlir/tosa/transforms/strip_complex_types.cc

```diff
@@ -0,0 +1,161 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// TOSA has no notion of complex datatypes, it represents a single complex
+// input tensor using two floating point input tensors corresponding to the
+// "real" and "imag" parts of the complex input.
+//
+// To maintain the correct mapping of these tensors during the legalization
+// from TFL to TOSA, a complex tensor of shape [x, ..., y] is converted to
+// a single floating point tensor of shape [x, ..., y, 2] where each resulting
```

Comment:
OOC does [x, 2] vs [2, x] matter here for further consumption? (wondering about slicing behavior)

Reply:
I don't believe so, as long as it's consistent. In the future we're planning to contribute a canonicalization of the concatenate->slice sequence that gets produced from consecutive complex operations, which will simplify the resulting TOSA graph quite a bit and remove unnecessary copying of data. After this, slice and concatenate operations should only exist on the inputs and outputs of the graph respectively.

We chose the [x, 2] format since this more closely resembles the format of a TFLite complex tensor, although there shouldn't be a problem with other frameworks adopting a different approach if necessary in the future.